html/browsers/the-window-object/BarProp.window.js

```diff
@@ -0,0 +1,53 @@
+function assert_barProps(barPropObjects, visible) {
+  let lastBarProp = undefined;
+  for (let barProp of barPropObjects) {
+    const currentBarProp = barProp;
+    assert_not_equals(currentBarProp, lastBarProp, "BarBrop objects of different properties are identical");
+    assert_equals(currentBarProp.visible, visible, "a BarProp's visible is wrong");
+    lastBarProp = currentBarProp;
+  }
+}
+
+function assert_identical_barProps(barProps, w, oldBarPropObjects, visible) {
+  barProps.map(val => w[val]).map((val, index) => {
+    assert_equals(val, oldBarPropObjects[index], "BarProp identity not preserved");
+  });
+  assert_barProps(oldBarPropObjects, visible);
+}
+
+async_test(t => {
+  const frame = document.body.appendChild(document.createElement("iframe")),
+        frameW = frame.contentWindow,
+        barProps = ["locationbar", "menubar", "personalbar", "scrollbars", "statusbar", "toolbar"],
+        barPropObjects = barProps.map(val => frameW[val]);
+
+  assert_barProps(barPropObjects, true);
+  frame.remove();
+  assert_identical_barProps(barProps, frameW, barPropObjects, false);
+  t.step_timeout(() => {
+    assert_identical_barProps(barProps, frameW, barPropObjects, false);
+    t.done();
+  }, 0);
+}, "BarBrop objects of a nested Window");
+
+async_test(t => {
+  const openee = window.open("/common/blank.html"),
+        barProps = ["locationbar", "menubar", "personalbar", "scrollbars", "statusbar", "toolbar"],
+        barPropObjects = barProps.map(val => openee[val]);
+
+  assert_barProps(barPropObjects, true);
+  openee.onload = t.step_func(() => {
+    assert_identical_barProps(barProps, openee, barPropObjects, true);
+
+    openee.onunload = t.step_func(() => {
+      assert_barProps(barPropObjects, true);
```

Comment:
Whenever you use `assert_barProps` instead of `assert_identical_barProps`, I wonder to myself: does this mean they're supposed to have changed since the last time? If so, replacing it with an assertion that they are supposed to be different would be good. If not, replacing it with `assert_identitical_barProps` would be good.